models.py

```diff
@@ -178,55 +178,67 @@ def empty_order(self):
         self.save()
         return self
 
-    def add_item(self, item, quantity=1):
+    def add_items(self, items):
         """
-        Add one item to the order.
-        Creates one OrderItem row if the item is not present in the order yet,
-        or increasing the count of the existing OrderItem.
-
-        :param item Item: instance of models.Item
+        Add items to the order from a dict {<Item>: <int:quantity>}. Handles
+        creating or updating the OrderItem cross table and the Order total
+        :param dict items: {<Item>: <quantity:int>}
         """
-
-        for orderitem in self.order_items:
-            # Looping all the OrderItem related to this order, if one with the
-            # same item is found we update that row.
-            if orderitem.item == item:
-                orderitem.add_item(quantity)
+        with database.atomic():
+            for (item, quantity) in items.items():
+                for orderitem in self.order_items:
```

Comment:
dopo controllo. mi sembra di averla messa dentro per gestire il caso in cui arrivi una richiesta (sbagliata) di aggiungere due volte lo stesso item.
se mi arriva qualcosa come `{ <Item1>: 2, <Item1>: 3 }` se la chiamata ad order_items è fuori dal loop vengono aggiunte due row in OrderItem, che puntano allo stesso oggetto dallo stesso ordine.